dot/core/service.go

```diff
@@ -370,23 +373,42 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 // the new block, revalidates the transactions in the pool, and moves
 // them to the queue if valid.
 // See https://github.com/paritytech/substrate/blob/74804b5649eccfb83c90aec87bdca58e5d5c8789/client/transaction-pool/src/lib.rs#L545
-func (s *Service) maintainTransactionPool(block *types.Block) {
+func (s *Service) maintainTransactionPool(block *types.Block) error {
 	// remove extrinsics included in a block
 	for _, ext := range block.Body {
 		s.transactionState.RemoveExtrinsic(ext)
 	}
 
+	bestBlockHash := s.blockState.BestBlockHash()
```

Comment:
instead of retrieving the best block hash inside this function, would be nice to pass it as an argument. I saw that in the caller context the best block hash is already known (`dot/core/service.go` line `283`)